_posts/izayak/2016-02-15-midterm-reflection-ry.md

```diff
@@ -0,0 +1,65 @@
+rather math way I think. I believe the way that Elliott is trying to tell is a more programming way with loops and conditionals.   
+I also tried the advanced features last Thursday:
+<iframe src="https://trinket.io/embed/python/b8cac28955" width="100%" height="600" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>  
+(In the previous Trinket, I changed all the 10 to width and length to make the program more flexible, not in this one)  
+For this one, I haven't figured out the default poem part. I don't know how to put the default poem into a list. The lines will remain   
+separately with quotes.  
+<iframe src="https://trinket.io/embed/python/3846dec449" width="100%" height="600" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>  
+Similar problems in this in-class exercise of chapter 8 (lists). I commented the top few lines, don't know if can combine the lines  
+together first. I don't know how to remove quotes in this case either. When lines are put together, I cannot do split later as in the way   
+shown in class. 
+
+For things done well and struggled with, as I mentioned before, I am better at concepts than doing real problems. It is both because of   
+trainings have been taken and personal preference. Especially with real problems, ususlly the division of problems into small tasks is more  
+required, which I am bad at. I know that the strategy should be do things step by step, but I always end up doing things at one time.  
+Sometimes I do them early, sometimes at last, both are not good experience. In the future, I will try to overcome it. And the strategy   
+I will continue using is searching online and trying to find resources by myself when facing some unknown concepts. Other people's ideas 
+are inspiring, and learning things ahead is a good way to encourage myself to learn more.  
+
+Working with partners, as well as facing unknown concepts, is what we will encounter in a future job. So it is important to try to get   
+used to it and find a way to collaborate better with classmates. Till now, I haven't figured out that, actually, but I do feel the   
+benefits sometimes. The problem is that we have different ideas and kind of did not really get the partner's idea. The pair programming   
+for novice like us can be slower, but getting insights from each other can be really helpful for opening minds. I think the key point is  
+that we should try to understand other people's idea and try to clearly convey oneself's idea, or we will just end up in saying oneself's  
+idea again and again without communication and improvements. If partners both think in their own way and try to do in their own way will  
+result in no idea exchanging. In that situation, pair work would be less efficient than working alone. Also I have question about the   
+job of navigator and driver, should the driver do whatever the navigator is thinking about or the idea discussed and agreed by both of   
```

Comment:
Good question.  The driver generally chooses *how* to do something, while the navigator focuses on *what* you should be doing to solve the problem.  But ideally it's a negotiation and give-and-take.